ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/clientlibs/editor/utils/utils.js

```diff
@@ -210,7 +210,49 @@
                         patternComponent.value="custom";
                     }
                 }
+            }
+        }
+
+        static registerDialogValueTypeValidators(dataTypeSelector, defaultTypeSelector, enumSelector) {
+            return function registerDialogValidators() {
+                var isBoolean = function(value) {
+                    var isBoolean = false;
+                    if (value) {
+                        var lowerCaseValue = value.toLowerCase();
+                        isBoolean = lowerCaseValue === 'true' || lowerCaseValue === 'false';
+                    }
+                    return isBoolean
+                }
+
+                function registerValidator(selector, validator) {
+                    $(window).adaptTo("foundation-registry").register("foundation.validation.validator", {
+                        selector: selector,
+                        validate: validator
+                    });
+                }
+
+                var dataTypeValidator = function(el) {
+                    var isValid = true;
+                    var value = el.value;
+                    var dataTypeLabel = document.querySelector(dataTypeSelector);
+                    if (dataTypeLabel) {
+                        var dataType = dataTypeLabel.innerText? dataTypeLabel.innerText.toLowerCase() : "";
```

Comment:
we should look for the exact value instead of the label while implementing business logic. label is a presentation property.